jerry-core/jmem/jmem-allocator.c

```diff
@@ -221,6 +233,9 @@ jmem_decompress_pointer (uintptr_t compressed_pointer) /**< pointer to decompres
 #if defined (ECMA_VALUE_CAN_STORE_UINTPTR_VALUE_DIRECTLY) && defined (JERRY_CPOINTER_32_BIT)
   JERRY_ASSERT (uint_ptr % JMEM_ALIGNMENT == 0);
 #else /* !ECMA_VALUE_CAN_STORE_UINTPTR_VALUE_DIRECTLY || !JERRY_CPOINTER_32_BIT */
+#ifndef JERRY_SYSTEM_ALLOCATOR
+  JERRY_DEFINE_CURRENT_CONTEXT ();
+#endif /* JERRY_SYSTEM_ALLOCATOR */
```

Comment:
I'm not sure about the necessity of this guard. If `JERRY_SYSTEM_ALLOCATOR` _is_ defined then `JERRY_HEAP_CONTEXT` is not available at all, i.e., the line below will not compile. So, here, I'm almost sure that `JERRY_SYSTEM_ALLOCATOR` is not defined, which means that the guard can be dropped.

Reply:
Without the guard, Travis gives the following error: 
`Variable 'jerry_current_context' is assigned a value that is never used.`
The guard helps to avoid this problem. I'm open to other solutions, but for the time, this seemed the best solution for me.